source/common/router/scoped_rds.cc

```diff
@@ -84,9 +84,7 @@ InlineScopedRoutesConfigProvider::InlineScopedRoutesConfigProvider(
                                                  ConfigProvider::ApiType::Delta),
       name_(std::move(name)),
       config_(std::make_shared<ScopedConfigImpl>(std::move(scope_key_builder))),
-      config_protos_(std::make_move_iterator(config_protos.begin()),
-                     std::make_move_iterator(config_protos.end())),
-      rds_config_source_(std::move(rds_config_source)) {}
+      config_protos_(move(config_protos)), rds_config_source_(std::move(rds_config_source)) {}
```

Comment:
Can you remove the unrelated optimization fixups and put them in a separate `misc:` PR?

Reply:
Yes, I agree that these changes should be in another PR.